src/App.tsx

```diff
@@ -1,61 +1,152 @@
 import './App.scss';
+import React, { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import users from './api/users';
+import todosFromServer from './api/todos';
+
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import { User } from './types/User';
+
+const findUserById = (userId: number): User | null => (
+  users.find(user => user.id === userId) || null
+);
+
+const usersTodoList: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: findUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState(usersTodoList);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [completed, setCompleted] = useState(false);
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+
+  const handleError = () => {
+    if (!title.trim().length) {
+      setTitleError(true);
+    }
+
+    if (!userId) {
+      setUserError(true);
+    }
+  };
+
+  const handleInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value.replace(/[^A-Za-z-А-Яа-я-іІїЇєЄ\s]/g, ''));
+    setTitleError(false);
+  };
+
+  const handleSelect = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setUserError(false);
+  };
+
+  const revertDefault = (todo: Todo) => {
+    setTodos([...todos, todo]);
+    setTitle('');
+    setUserId(0);
+    setCompleted(false);
+  };
+
+  const onSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!title.trim().length || !userId) {
+      handleError();
+
+      return;
+    }
+
+    const id = Math.max(...todos.map(todo => todo.id)) + 1;
+    const user = users.find(u => u.id === userId) || null;
+
+    const newTodo = {
+      id,
+      title,
+      completed,
+      userId,
+      user,
+    };
+
+    revertDefault(newTodo);
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={event => onSubmit(event)}
```

Comment:
No need for wrapper if you just pass event object.
```suggestion
        onSubmit={onSubmit}
```